amqp.go

```diff
@@ -601,13 +631,32 @@ func (c *Client) SetupPublishing(exchange string) error {
 	return err
 }
 
-// Publish sends a message to the configured exchange with a routing key set to
-// the value of 'key'.
-func (c *Client) Publish(key string, body []byte) error {
+// PublishingOpts contains a set of options for publishing AMQP messages.
+type PublishingOpts struct {
+	DeliveryMode uint8
+	ContentType  string
+}
+
+// DefaultPublishingOpts defines the set of publishing options used by default.
+var DefaultPublishingOpts = &PublishingOpts{
+	DeliveryMode: amqp.Persistent,
+	ContentType:  "text/plain",
+}
+
+// JSONPublishingOpts defines the set of publishing options used for JSON message
+// bodies.
+var JSONPublishingOpts = &PublishingOpts{
+	DeliveryMode: amqp.Persistent,
+	ContentType:  "application/json",
+}
+
+// PublishOpts sends a message to the configured exchange with options specified
+// in an options structure.
+func (c *Client) PublishOpts(key string, body []byte, opts *PublishingOpts) error {
```

Comment:
The current notification agent uses a content type of `application/json` when publishing messages intended for consumption by the UI. This change allows callers to specify the content type or delivery mode without too much hassle.